Dotnet_New_Template_Restore_Success(template: "webapiaot") is failing on Linux

Linux agents started experiencing this error when creating a new Web AOT app in our integration tests:

```
> /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/dotnet new webapiaot
│  The template "ASP.NET Core Web API (native AOT)" was created successfully.
│  
│  Processing post-creation actions...
│  Restoring /home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj:
│  
│    Determining projects to restore...
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018: The "ProcessFrameworkReferences" task failed unexpectedly. [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018: System.ArgumentNullException: Value cannot be null. (Parameter 'key') [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.FindValue(TKey key) [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.ContainsKey(TKey key) [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018:    at Microsoft.NET.Build.Tasks.NuGetUtils.GetBestMatchingRidWithExclusion(RuntimeGraph runtimeGraph, String runtimeIdentifier, IEnumerable`1 runtimeIdentifiersToExclude, IEnumerable`1 availableRuntimeIdentifiers, Boolean& wasInGraph) [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ProcessFrameworkReferences.AddToolPack(ToolPackType toolPackType, Version normalizedTargetFrameworkVersion, List`1 packagesToDownload, List`1 implicitPackageReferences) [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ProcessFrameworkReferences.ExecuteCore() [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Execute() [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  /home/vsts/work/1/s/.test/work/d0c9c47d/0918cfbb/sdk/10.0.100-rc.2.25454.107/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(119,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(TaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/vsts/work/1/s/.test/work/1c5f60df/d2926dc4/solution/solution.csproj]
│  Restore failed.
│  Post action failed.
│  Manual instructions: Run 'dotnet restore'
│  
│  
└ Completed in 1.17s
```

--- a/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs
+++ b/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs
@@ -1739,7 +1739,7 @@ EndGlobal";
         [InlineData("nunit")]
         [InlineData("web")]
         [InlineData("webapi")]
-        [InlineData("webapiaot")]
+        // [InlineData("webapiaot")] Skipping due to failures: https://github.com/NuGet/Home/issues/14523
         [InlineData("webapp")]
         [InlineData("worker")]
         [InlineData("xunit")]

